source/_posts/2021-12-21-android-december-2021-release.markdown

```diff
@@ -0,0 +1,129 @@
+
+
+## Theme and UI Updates
+
+In this release we had a lot of changes being done to the overall theme of the app to better fit with the design of the Home Assistant frontend theme. The status and navigation bar will now match your theme of choice. The overall loading experience has also had some improvements to align more closely to the browser loading experience. Thank you to [LasseRosenow] for all your hard work here!
+
+With the release of [Jetpack Compose] we have decided to start migrating all UI elements to Compose. If you are familiar with Android Development then you will remember that the UI is always built with XML and then referenced in your activities/fragments. Now with Compose, XML is no longer needed and bulding robust UI's becomes a breeze. We find these new libraries to be very easy to use and it has allowed us to improve upon our internal architecture to make things easier for new and upcoming features.
+
+In the phone app the entire onboarding experience has been rewritten in Compose, including a brand new welcome screen to help first time users understand what Home Assistant is all about. The notification detail page found in notification history has also received a compose update. The Wear OS home screen is actually built using compose including the new settings screens found in the phone app.
+
+<p class='img'>
+<img src='/images/blog/2021-12-21-android-december-2021/welcome.png' alt='Screenshot of welcome screen'>
+Screenshot of welcome screen
+</p>
+
+## Other Changes
+
+With so many changes since the last update its impossible to list all of the other cool new features but here is a list of some welcomed improvements:
+
+*  New sensor to report the state of [High Accuracy Mode] by [dshokouhi]
+*  New notification parameters to change the [Status Bar Icon] and also to [alert once] for any given notification by [dshokouhi]
+*  New [WebView] settings to keep the screen on and to auto play videos by [dshokouhi] and [skynetua]
+*  New [notification command] to keep the screen on by [skynetua]
+*  Updated notification commands to accept URL encoding in extras by [mvn23]
+*  [BLE Transmitter] improvements to power output and adverister mode by [Alfiegerner] and [amadeo-alex]
+*  Quick Settings Tile limit increased from 5 to 12 by [dshokouhi]
+*  Support for Android 12 by [dshokouhi]
+*  Updated design for Media Player Widget by [jannis3005]
+
+<p class='img'>
+<img src='/images/blog/2021-12-21-android-december-2021/media_player.png' alt='Screenshot of Media Player Widget'>
+Screenshot of Media Player Widget
+</p>
+
+*  Support for cookie based authentication by [duncf]
+*  Setting to always try the internal URL first. This is helpful to those who like to leave location off by [dshokouhi]
```

Comment:
Sentence is cut off.

```suggestion
*  Setting to always try the internal URL first. This is helpful to those who like to leave location off by [dshokouhi]
```

Reply:
Dont think there was a change in the suggestion here? It doenst look cut off to me on my end :)